packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -1116,15 +1126,21 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with Restoratio
   }
 
   Widget _buildClearButton() {
-    return GestureDetector(
-      key: _clearGlobalKey,
-      onTap: widget.enabled ? _onClearButtonTapped : null,
-      child: Padding(
-        padding: const EdgeInsets.symmetric(horizontal: 6.0),
-        child: Icon(
-          CupertinoIcons.clear_thick_circled,
-          size: 18.0,
-          color: CupertinoDynamicColor.resolve(_kClearButtonColor, context),
+    final CupertinoLocalizations localizations = CupertinoLocalizations.of(context);
```

Comment:
I would do:

```dart
final String clearLabel = widget.clearButtonSemanticLabel ?? CupertinoLocalizations.of(context).clearButtonLabel;
```
That way, you only establish a dependency on CupertinoLocalizations if you actually need it.